### ISSUE TITLE
Incorrect references.special_chars when using Snowflake metadata column

### ISSUE THREAD

**Issue by rogalski** [sqlfluff/sqlfluff#5358]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

When linting valid Snowflake code, following violation is raised incorrectly.


### Expected Behaviour

Since we only select from invalid named column, not creating invalid named column, violation shouldn't be raised.

### Observed Behaviour

```
== [bug1.sql] FAIL                                                                                                                                                                                                           
L:   2 | P:   5 | RF05 | Do not use special characters in identifiers.
                       | [references.special_chars]
```

### How to reproduce

```sql
SELECT
    METADATA$FILENAME AS FILENAME,
    SOME_COLUMN
FROM
    @EXT_STAGE/path/to/file
    (FILE_FORMAT => MY_FORMAT);
```

### Dialect

snowflake

### Version

sqlfluff, version 2.3.4
python 3.10.12

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rogalski** [comment]
#5358 and #5395 are likely same underlying root cause.
